src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1408,99 +1409,185 @@ static constexpr bool _isHexNumber(const wchar_t wch) noexcept
+                }
+
+                parameterValue += intVal;
+
+                if (isSharpSignFormat)
+                {
+                    // If we get this far, any number can be seen as a valid part
+                    // of this component.
+                    foundColor = true;
+
+                    if (i >= rgbHexDigitCount)
                     {
-                        value += intVal;
+                        // Successfully parsed this component. Start the next one.
+                        break;
                     }
-                    else
+                }
+                else
+                {
+                    // Record the hex digit count of the current component.
+                    rgbHexDigitCount = i + 1;
+
+                    // If this is the first 2 component...
+                    if (component < 2 && curr < string.cend() && *curr == L'/')
                     {
-                        // Encountered something weird oh no
-                        foundColor = false;
+                        // ...and we have successfully parsed this component, we need
+                        // to skip the delimiter before starting the next one.
+                        curr++;
+                        foundColor = true;
                         break;
                     }
-                    // If we're on the blue component, we're not going to see a /.
-                    // Break out once we hit the end.
-                    if (component == 2 && curr >= string.cend())
+                    // Or we have reached the end of the string...
+                    else if (curr >= string.cend())
                     {
-                        foundValidColorSpec = true;
+                        // ...meaning that this is the last component. We're not going to
+                        // see any delimiter. We can just break out.
+                        foundColor = true;
                         break;
                     }
                 }
-                else if (wch == L'/')
-                {
-                    // Break this component, and start the next one.
-                    foundColor = true;
-                    break;
-                }
-                else
-                {
-                    // Encountered something weird oh no
-                    foundColor = false;
-                    break;
-                }
             }
-            if (!foundColor || curr >= string.cend())
+
+            if (!foundColor)
             {
-                // Indicates there was a some error parsing color
-                //  or we're at the end of the string.
-                break;
+                // Indicates there was some error parsing color.
+                return false;
             }
+
+            // Calculate the actual color value based on the hex digit count.
+            auto& colorValue = til::at(colorValues, component);
+            const auto scaleMultiplier = 0x11;

```

Comment:
@j4james So I notice that you suggested me to use

```cpp
const auto scaleMultiplier = isSharpSignFormat ? 0x10 : 0x11;
```

But I choose `scaleMultiplier = 0x11` to have the same behaviour as VTE, as described in Note 2 in #3715. First I don't want to add even more logic to this already stacked method. Second it's closer to what's popular on the web. I think it's a valuable deviation from the standard.

Reply:
I'm strongly opposed to this. We don't get to adopt somebody else's sequence and then redefine what it means. The fact that VTE implemented it incorrectly is no reason to copy them. The idea that it was done that way to match CSS is just a post hoc rationalization for their bug. They used a built in a API for the color conversion that just happened to work the same way as CSS - if it had matched the XTerm spec instead, I guarantee you they wouldn't have tried to change it.

Reply:
I have to agree here; if we're going to pretend to be xterm-256color, we should act as close to it as we can. I know that at the boundaries of compatibility this will cause applications to act different from expectations, but only very slightly.

At the end of the day, if somebody files a bug on us saying that `#fff` acts like `#f0f0f0` instead of `#ffffff`, we have a [documented standard to fall back on](https://linux.die.net/man/3/xparsecolor).

Reply:
As much I like the idea of using CSS format, I respect you guys even more. I'll change this.